src/main/java/uk/gov/justice/digital/hmpps/whereabouts/listeners/SqsDomainEventListener.kt

```diff
@@ -15,29 +16,36 @@ class SqsDomainEventListener(
   @Qualifier("videoLinkBookingServiceAppScope")
   private val videoLinkBookingService: VideoLinkBookingService,
   private val gson: Gson,
+  @Value("\${feature.bvls.enabled}") private val bvlsEnabled: Boolean,
 ) {
   companion object {
-    val log: Logger = LoggerFactory.getLogger(this::class.java)
+    private val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  init {
+    log.info("BVLS enabled=$bvlsEnabled for domain events")
   }
 
   @SqsListener("domainevent", factory = "hmppsQueueContainerFactoryProxy")
   @WithSpan(value = "map-devs-hmpps_whereabouts_api_domain_queue", kind = SpanKind.SERVER)
   fun handleDomainEvents(requestJson: String?) {
-    try {
-      log.info("Raw domain event message: {}", requestJson)
-      val (message, messageAttributes) = gson.fromJson(requestJson, Message::class.java)
-      val eventType = messageAttributes.eventType.value
-      log.info("Processing message of type {}", eventType)
+    if (bvlsEnabled) {
+      try {
+        log.info("Raw domain event message: {}", requestJson)
+        val (message, messageAttributes) = gson.fromJson(requestJson, Message::class.java)
+        val eventType = messageAttributes.eventType.value
+        log.info("Processing message of type {}", eventType)
 
-      when (eventType) {
-        "prison-offender-events.prisoner.released" -> {
-          val transferEventMessage = gson.fromJson(message, ReleasedOffenderEventMessage::class.java)
-          videoLinkBookingService.deleteAppointmentWhenTransferredOrReleased(transferEventMessage)
+        when (eventType) {
+          "prison-offender-events.prisoner.released" -> {
+            val transferEventMessage = gson.fromJson(message, ReleasedOffenderEventMessage::class.java)
+            videoLinkBookingService.deleteAppointmentWhenTransferredOrReleased(transferEventMessage)
+          }
         }
+      } catch (e: Exception) {
+        log.error("processDomainEvent() Unexpected error", e)
+        throw e
       }
```

Comment:
Did we want to log that the event has been received and ignored, or just not mention it?

Reply:
I did ponder this very thought and went with no.  Happy to log if you feel strongly about it.  I did add a log message to the init so we at least know it is enabled or not.